pkg/es/config/config.go

```diff
@@ -33,48 +33,55 @@ import (
-	BulkSize                       int            `mapstructure:"-"`
-	BulkWorkers                    int            `mapstructure:"-"`
-	BulkActions                    int            `mapstructure:"-"`
-	BulkFlushInterval              time.Duration  `mapstructure:"-"`
-	IndexPrefix                    string         `mapstructure:"index_prefix"`
-	IndexDateLayoutSpans           string         `mapstructure:"-"`
-	IndexDateLayoutServices        string         `mapstructure:"-"`
-	IndexDateLayoutSampling        string         `mapstructure:"-"`
-	IndexDateLayoutDependencies    string         `mapstructure:"-"`
-	IndexRolloverFrequencySpans    string         `mapstructure:"-"`
-	IndexRolloverFrequencyServices string         `mapstructure:"-"`
-	IndexRolloverFrequencySampling string         `mapstructure:"-"`
-	ServiceCacheTTL                time.Duration  `mapstructure:"service_cache_ttl"`
-	AdaptiveSamplingLookback       time.Duration  `mapstructure:"-"`
-	Tags                           TagsAsFields   `mapstructure:"tags_as_fields"`
-	Enabled                        bool           `mapstructure:"-"`
-	TLS                            tlscfg.Options `mapstructure:"tls"`
-	UseReadWriteAliases            bool           `mapstructure:"use_aliases"`
-	CreateIndexTemplates           bool           `mapstructure:"create_mappings"`
-	UseILM                         bool           `mapstructure:"use_ilm"`
-	Version                        uint           `mapstructure:"version"`
-	LogLevel                       string         `mapstructure:"log_level"`
-	SendGetBodyAs                  string         `mapstructure:"send_get_body_as"`
+	Servers                  []string       `mapstructure:"server_urls" valid:"required,url"`
+	RemoteReadClusters       []string       `mapstructure:"remote_read_clusters"`
+	Username                 string         `mapstructure:"username"`
+	Password                 string         `mapstructure:"password" json:"-"`
+	TokenFilePath            string         `mapstructure:"token_file"`
+	PasswordFilePath         string         `mapstructure:"password_file"`
+	AllowTokenFromContext    bool           `mapstructure:"-"`
+	Sniffer                  bool           `mapstructure:"sniffer"` // https://github.com/olivere/elastic/wiki/Sniffing
+	SnifferTLSEnabled        bool           `mapstructure:"sniffer_tls_enabled"`
+	MaxDocCount              int            `mapstructure:"-"` // Defines maximum number of results to fetch from storage per query
+	MaxSpanAge               time.Duration  `mapstructure:"-"` // configures the maximum lookback on span reads
+	Timeout                  time.Duration  `mapstructure:"-"`
+	BulkSize                 int            `mapstructure:"-"`
+	BulkWorkers              int            `mapstructure:"-"`
+	BulkActions              int            `mapstructure:"-"`
+	BulkFlushInterval        time.Duration  `mapstructure:"-"`
+	IndexPrefix              string         `mapstructure:"index_prefix"`
```

Comment:
This is confusing and inconsistent with the rest of the parameters. Either prefix is unique per index or it's shared by all indices. You have both, what does it mean?